File: api/src/schemas/pf_diameters.py
from typing import Optional, List
from pydantic import BaseModel, Field
from datetime import datetime

class PFDiameterBase(BaseModel):
    value: float = Field(..., description="管徑值")
    name: str = Field(..., max_length=50, description="管徑名稱 (例如 DN50, 2吋)")
    # description: Optional[str] = Field(None, description="管徑描述") # 可選

class PFDiameterCreate(PFDiameterBase):
    pass

class PFDiameterUpdate(BaseModel):
    value: Optional[float] = Field(None, description="管徑值")
    name: Optional[str] = Field(None, max_length=50, description="管徑名稱")
    # description: Optional[str] = Field(None, description="管徑描述") # 可選

class PFDiametersResponse(PFDiameterBase):
    id: int = Field(..., description="管徑ID")
    # 如果您的模型自動管理 created_at 和 modified_at，可以取消註釋
    # created_at: datetime
    # modified_at: datetime

    class Config:
        from_attributes = True # For Pydantic V2

class PFDiametersListResponse(BaseModel):
    items: List[PFDiametersResponse]
    total: int

File: dry-farm/src/types/pfDiameters.ts
// 口徑基本接口
export interface PFDiameterBase {
  value: number;
  name: string;
}

// 用於創建新口徑的接口
export interface PFDiameterCreate extends PFDiameterBase {
}

// 用於更新口徑的接口
export interface PFDiameterUpdate {
  value?: number;
  name?: string;
}

// 口徑響應接口
export interface PFDiameter {
  id: number;
  value: number;
  name: string;
}

// 分頁響應接口
export interface PaginatedResponse<T> {
  items: T[];
  total: number;
}
